File: types/risk-evaluation.ts
export type RiskLevel = "BAJO" | "MEDIO" | "ALTO"

export type EvaluationType = "INICIAL" | "PERIODICA" | "RE_EVALUACION"

export type EvaluationStatus = "DRAFT" | "PENDING_APPROVAL" | "APPROVED" | "REJECTED"

export interface RiskFactor {
  factorId: string
  factorName: string
  description: string
  weight: number // 0-5
  observation: string
  requiresObservation?: boolean
}

export interface RiskEvaluation {
  evaluationId: string
  dossierId: string
  dossierName?: string
  evaluationType: EvaluationType
  evaluationDate: string
  evaluatorUserId: string
  evaluatorName?: string
  version: number
  status: EvaluationStatus
  riskFactors: RiskFactor[]
  preliminaryRiskLevel: RiskLevel
  finalRiskLevel: RiskLevel
  totalScore: number
  hasManualOverride: boolean
  manualOverrideJustification?: string
  overrideAppliedBy?: string
  requiresApproval: boolean
  approvedBy?: string
  approvedAt?: string
  rejectionReason?: string
  nextReviewDate?: string
  comments?: string
  createdAt: string
  updatedAt?: string
}

export interface CreateEvaluationRequest {
  dossierId: string
  evaluationType: EvaluationType
  riskFactors: RiskFactor[]
  userId: string
  comments?: string
}

export interface UpdateEvaluationRequest {
  evaluationId: string
  riskFactors: RiskFactor[]
  userId: string
}

export interface ApproveEvaluationRequest {
  evaluationId: string
  approvalComments?: string
  userId: string
}

export interface RejectEvaluationRequest {
  evaluationId: string
  rejectionReason: string
  userId: string
}

File: backend/src/main/java/com/siar/risk/model/RiskEvaluation.java
package com.siar.risk.model;

import com.fasterxml.jackson.databind.JsonNode;
import com.vladmihalcea.hibernate.type.json.JsonBinaryType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Type;
import org.hibernate.annotations.TypeDef;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "risk_evaluations", indexes = {
    @Index(name = "idx_dossier_id", columnList = "dossierId"),
    @Index(name = "idx_status", columnList = "status"),
    @Index(name = "idx_evaluation_date", columnList = "evaluationDate"),
    @Index(name = "idx_final_risk_level", columnList = "finalRiskLevel"),
    @Index(name = "idx_next_review_date", columnList = "nextReviewDate")
})
@TypeDef(name = "jsonb", typeClass = JsonBinaryType.class)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RiskEvaluation {
    
    @Id
    @Column(length = 50)
    private String evaluationId;  // EVAL-2024-000123-v1
    
    @Column(nullable = false, length = 50)
    private String dossierId;  // FK to Dossier
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EvaluationType evaluationType;
    
    @Column(nullable = false)
    private LocalDateTime evaluationDate;
    
    @Column(nullable = false, length = 50)
    private String evaluatorUserId;  // FK to User
    
    @Column(nullable = false)
    private Integer version;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private EvaluationStatus status;
    
    @Type(type = "jsonb")
    @Column(nullable = false, columnDefinition = "jsonb")
    private JsonNode riskFactorsJson;
    
    @Type(type = "jsonb")
    @Column(columnDefinition = "jsonb")
    private JsonNode calculationResultJson;
    
    @Column(length = 50)
    private String configurationId;  // FK to RiskConfiguration
    
    @Enumerated(EnumType.STRING)
    @Column(length = 10)
    private RiskLevel preliminaryRiskLevel;
    
    @Enumerated(EnumType.STRING)
    @Column(length = 10)
    private RiskLevel finalRiskLevel;
    
    @Column(nullable = false)
    private Boolean hasManualOverride;
    
    @Column(columnDefinition = "TEXT")
    private String manualOverrideJustification;
    
    @Column(length = 50)
    private String overrideAppliedBy;  // FK to User
    
    private LocalDateTime overrideAppliedAt;
    
    @Column(nullable = false)
    private Boolean requiresEnhancedDueDiligence;
    
    @Column(nullable = false)
    private Boolean requiresApproval;
    
    @Column(length = 30)
    private String approvalLevel;
    
    @Column(length = 50)
    private String approvedBy;  // FK to User
    
    private LocalDateTime approvedAt;
    
    @Column(columnDefinition = "TEXT")
    private String rejectionReason;
    
    private LocalDate nextReviewDate;
    
    @Column(nullable = false)
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @Column(columnDefinition = "TEXT")
    private String comments;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (hasManualOverride == null) hasManualOverride = false;
        if (requiresEnhancedDueDiligence == null) requiresEnhancedDueDiligence = false;
        if (requiresApproval == null) requiresApproval = true;
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
